### PR TITLE
JavaCrossCompilePlugin: use CompileOptions release property

### PR DESCRIPTION
**Pull request by rpalcolea**

### CODE REVIEW

File: src/integTest/groovy/nebula/plugin/compile/JavaCrossCompilePluginIntegrationSpec.groovy [hunk 1]
```diff
@@ -124,8 +124,8 @@ class JavaCrossCompilePluginIntegrationSpec extends IntegrationSpec {
         writeHelloWorld('helloworld')
 
         when:
-        runTasks('--configuration-cache=on', 'compileJava')
-        def result = runTasks('--configuration-cache=on', 'compileJava')
+        runTasks('--configuration-cache', 'compileJava')
+        def result = runTasks('--configuration-cache', 'compileJava')
```
Comment (rpalcolea) [inline src/integTest/groovy/nebula/plugin/compile/JavaCrossCompilePluginIntegrationSpec.groovy:128]: feature flag for configuration cache changed between Gradle 6.5 and 6.6

File: src/main/kotlin/nebula/plugin/compile/JavaCrossCompilePlugin.kt [hunk 2]
```diff
@@ -45,7 +46,11 @@ class JavaCrossCompilePlugin @Inject constructor(private val providerFactory: Pr
                 val location by lazy { targetCompatibility.locate(project, providers) }
                 withType(JavaCompile::class.java) {
                     if (JavaVersion.current() >= JavaVersion.VERSION_1_9) {
-                        it.options.compilerArgs.addAll(listOf("--release", targetCompatibility.majorVersion))
+                        if (project.gradle.versionLessThan("6.6-rc-1")) {
+                            it.options.compilerArgs.addAll(listOf("--release", targetCompatibility.majorVersion))
+                        } else {
+                            it.options.release.set(targetCompatibility.majorVersion.toInt())
```
Comment (rpalcolea) [inline src/main/kotlin/nebula/plugin/compile/JavaCrossCompilePlugin.kt:52]: use `CompileOptions.release` property with Gradle 6.6-rc-1 +

Comment (OdysseusLives) [reply src/main/kotlin/nebula/plugin/compile/JavaCrossCompilePlugin.kt:52]: Nice!

Comment (OdysseusLives) [reply src/main/kotlin/nebula/plugin/compile/JavaCrossCompilePlugin.kt:52]: The updated docs, for our future information: [6.6-rc-2 CompileOptions](https://docs.gradle.org/6.6-rc-2/dsl/org.gradle.api.tasks.compile.CompileOptions.html)

**Pull request merged**